### PR TITLE
OAM Component for a CDN

### PR DESCRIPTION
**Pull request by 5herlocked**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: platform/components/cdn.cue [hunk 1]
```diff
@@ -0,0 +1,144 @@
+"cloudfront-distribution": {
+	alias: "cdn"
+	annotations: {}
+	attributes: workload: definition: {
```
Comment (5herlocked) [inline platform/components/cdn.cue:4]: CRD DNE, just use K8s/Crossplane definitions @zjaco13 uses Deployments.

File: platform/components/cdn.cue [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	output: {
+		apiVersion: "aws.crossplane.io/v1beta1"
+		kind:       "Composition"
+		metadata: {
+			name: "s3-cloudfront-distribution"
+		}
+		spec: {
+			compositeTypeRef: {
+				apiVersion: "aws-service/v1"
+				kind:       "CDN"
```
Comment (5herlocked) [inline platform/components/cdn.cue:26]: Look at comment above

File: platform/components/cdn.cue [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+						apiVersion: "cloudfront.aws.crossplane.io/v1alpha1"
+						kind:       "Distribution"
+						spec: {
+							forProvider: {
+								distributionConfig: {
+									enabled:       true
+									httpVersion:   "http2"
+									isIPV6Enabled: true
+									origins: [{
+										domainName: "\($.resources[0].status.atProvider.regionDomainName)"
```
Comment (5herlocked) [inline platform/components/cdn.cue:72]: Zach says it no work.
First test, then find workaround.

File: platform/components/cdn.cue [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+										id:         "S3Origin"
+										s3OriginConfig: {
+											originAccessIdentity: "\($.resources[1].status.atProvider.id)"
```
Comment (5herlocked) [inline platform/components/cdn.cue:75]: comment above

File: platform/components/cdn.cue [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+				},
+				{
+					name: "s3-bucket-policy"
+					base: {
+						apiVersion: "s3.aws.crossplane.io/v1beta1"
+						kind:       "BucketPolicy"
+						spec: {
+							forProvider: {
+								region:     parameter.region
+								bucketName: "\($.resources[0].status.atProvider.name)"
```
Comment (5herlocked) [inline platform/components/cdn.cue:119]: look above

File: platform/components/cdn.cue [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+								policy: json.Marshal({
+									Version: "2008-10-17"
+									Id:      "PolicyForCloudFrontPrivateContent"
+									Statement: [
+										{
+											Sid:    "AllowCloudFrontServicePrincipal"
+											Effect: "Allow"
+											Principal: Service: "cloudfront.amazonaws.com"
+											Action:   "s3:GetObject"
+											Resource: "arn:aws:s3:::\($.resources[0].status.atProvider.name)/*"
```
Comment (5herlocked) [inline platform/components/cdn.cue:129]: look above

File: platform/components/dynamodb.cue [hunk 7]
```diff
@@ -0,0 +1,48 @@
+"dynamodb-table": {
+	alias: "table"
+	annotations: {}
+	attributes: workload: definition: {
+		apiVersion: "aws-service/v1"
```
Comment (5herlocked) [inline platform/components/dynamodb.cue:5]: this no worky

File: platform/components/dynamodb.cue [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+		kind:       "Table"
+	}
+	description: "AWS DynamoDB table"
+	labels: {}
+	parameter: {
+		attributeDefinitions: [...#AttributeDefinition]
+		keySchema: [...#KeySchema]
+		tableName: string
+	}
```
Comment (5herlocked) [inline platform/components/dynamodb.cue:14]: `parameter.region` and `parameter.name` missing

File: platform/components/dynamodb.cue [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+		kind:       "Table"
+		metadata: {
+			name: paramter.name
+		}
+		spec: {
+			forProvider: {
+				region:               parameter.region
+				name:                 paramater.tableName
+				billingMode:          "PAY_PER_REQUEST"
+				attributeDefinitions: parameter.attributeDefinition
```
Comment (5herlocked) [inline platform/components/dynamodb.cue:40]: EoL needs `s`

File: deployment/examples/cdn.example.yaml [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+spec:
+  components:
+    - name: web-cdn
+      type: cdn
+      properties:
+        region: us-east-1
+    - name: web-server
+      type: webservice
+      properties:
+        image: nginx:latest
```
Comment (shapirov103) [inline deployment/examples/cdn.example.yaml:14]: I don't see how you deploy this. Requires CI that will copy files to S3. Then deployment can work through CDN.

Comment (shapirov103) [inline deployment/examples/cdn.example.yaml:14]: Do we need nginx component in the picture? Seems redundant/

**Pull request merged**